packages/material-ui-lab/src/ClockPicker/ClockPointer.tsx

```diff
@@ -1,49 +1,82 @@
 import * as React from 'react';
 import clsx from 'clsx';
-import { StyleRules, MuiStyles, withStyles, WithStyles } from '@material-ui/core/styles';
+import { experimentalStyled } from '@material-ui/core/styles';
+import {
+  unstable_composeClasses as composeClasses,
+  generateUtilityClass,
+  generateUtilityClasses,
+} from '@material-ui/unstyled';
 import { ClockView, CLOCK_WIDTH, CLOCK_HOUR_WIDTH } from './shared';
 
-export type ClockPointerClassKey = 'root' | 'animateTransform' | 'thumb' | 'noPoint';
+export type ClockPointerClassKey = keyof typeof clockPointerClasses;
 
-export const styles: MuiStyles<ClockPointerClassKey> = (
-  theme,
-): StyleRules<ClockPointerClassKey> => ({
-  root: {
-    width: 2,
-    backgroundColor: theme.palette.primary.main,
-    position: 'absolute',
-    left: 'calc(50% - 1px)',
-    bottom: '50%',
-    transformOrigin: 'center bottom 0px',
-  },
-  animateTransform: {
-    transition: theme.transitions.create(['transform', 'height']),
-  },
-  thumb: {
-    width: 4,
-    height: 4,
-    backgroundColor: theme.palette.primary.contrastText,
-    borderRadius: '50%',
-    position: 'absolute',
-    top: -21,
-    left: `calc(50% - ${CLOCK_HOUR_WIDTH / 2}px)`,
-    border: `${(CLOCK_HOUR_WIDTH - 4) / 2}px solid ${theme.palette.primary.main}`,
-    boxSizing: 'content-box',
-  },
-  noPoint: {
-    backgroundColor: theme.palette.primary.main,
-  },
-});
-
-export interface ClockPointerProps
-  extends React.HTMLProps<HTMLDivElement>,
-    WithStyles<typeof styles> {
+export interface ClockPointerProps extends React.HTMLAttributes<HTMLDivElement> {
+  classes?: typeof clockPointerClasses;
   hasSelected: boolean;
   isInner: boolean;
   type: ClockView;
   value: number;
 }
 
+export function getClockPointerUtilityClass(slot: string) {
+  return generateUtilityClass('MuiInternalClockPointer', slot);
+}
+
+export const clockPointerClasses = generateUtilityClasses('MuiInternalClockPointer', [
```

Comment:
We decided not to add the classes on the internal components too, unless they are already used somewhere. Is this the case, or is this leftover?

Reply:
yep, `classes` is not used anywhere so will remove it.